### DISCUSSION TITLE
Trip logger

### DISCUSSION THREAD

**Discussion by penroseg** [BimmerGestalt/AAIdrive#437] [Ideas]
Is there any possibility of adding in a trip logger similar to what was previously available in the Connected Drive app?  The Connected Drive app updates at the end of a journey (location, fuel, range, odometer etc).  Would it be possible to capture these into a table storage type array and then add in the other information you already seem to access when connected to the car (Efficiency Stats, Driving Style, Eco Range won, Oil Temp, Engine Temp etc) to show a trip log?  It could then calculate the trip distance from the previous to the current Odometer reading.  It could even be possible to extend this by adding in additional information which _may_ be available from the car such as temperature or other weather information.  Ideally, these could potentially be viewed in the app, but primarily, be available as a tabular file eg CSV which could be downloaded.

(NB: one small bug to note too.  The _Fuel Level_ variable on the Car Information screen is currently showing in gal (UK) and does not follow the setting from the car, which in my case is Litres.)

Finally, and less importantly (for me anyway), it would be great if we could capture more frequent location updates and build a GPX journey log.  Not sure if you have access to the updating current location when connected to the car or only when stopped.  [edit: I just read another thread here about background refresh and I _think_ that it updates every 10 seconds while not in focus and real-time if the Car Information screen is accessed]

**Comment by hufman** [comment]
This is a great idea! It was one of the first ideas for [an Addon app](https://github.com/BimmerGestalt/IDriveConnectAddons), but I'm slow to start because it has the potential to grow into a very large project: Each of these features has it's own large decision tree, which makes it intimidating to build (Which columns to log in CSV? What cloud provider for uploading, realtime streaming somewhere? Fuel tripometer tracking? Map visualization of the trips?). I'm hoping that someone more skilled in app development will learn from the example apps showing how to access the car data and be inspired to build a great app for this feature.

You did read correctly, the car data values that are remembered when disconnected from the car are only updated every 10 seconds unless the Car Information tab is currently visible. However, the Addon system providing car data to other apps will automatically subscribe for faster updates when apps request data. I built a proof-of-concept racing logger as a [plugin for the AndrOBD app](https://github.com/BimmerGestalt/IDriveConnectAddons/tree/main/androbd_gestalt) building on top of its CSV logging functionality, but the same idea would work for recording a GPX file too. (The cloud API indeed only updates the position when the car is stopped, but the local Apps connection provides realtime updates)

As for your Fuel Level, that's very odd! You should install the CDS Details addon and scroll down to the `vehicle.units` row and see if these units change as you toggle the car's settings. The code sets up the value conversion and the label based on the `fuel` setting, which means "gal (UK)" if the [setting is 2](https://github.com/BimmerGestalt/AAIdrive/blob/165d64084a0045e7df0c2f07d14b0820fdf6412d/app/src/main/java/me/hufman/androidautoidrive/carapp/CDSValues.kt#L108). This could be incorrect, based on a flawed understanding of the decompiled app code I used, so please check!

**Reply by plugfir3** [reply]
My car does not have connecteddrive, so I would love to see trip logger in home assistant. Is it possible to add some functions like mqtt to send data to record in home assistant?

I tried to install androbd plugin and use its mqtt plugin, but I couldn’t load data in androbd at all.
